ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -229,6 +229,111 @@ def test_delete_with_nonexistent_hostgroup(self, automember_hostgroup,
                 reason=u'%s: Automember rule not found' % hostgroup1.cn)):
             command()
 
+    def test_create_with_existent_group(self, automember_group, group1):
+        """ Try to add a rule that already exists """
+        group1.ensure_exists()
+        automember_group.ensure_exists()
+        command = automember_group.make_create_command()
+        with raises_exact(errors.DuplicateEntry(
+                reason=u'Automember rule with name "%s" '
+                       u'already exists' % group1.cn)):
+            command()
+
+    def test_create_with_existent_hostgroup(self, automember_hostgroup,
+                                            hostgroup1):
+        """ Try to add a rule with non-existent group """
+        hostgroup1.ensure_exists()
+        automember_hostgroup.ensure_exists()
+        command = automember_hostgroup.make_create_command()
+        with raises_exact(errors.DuplicateEntry(
+                reason=u'Automember rule with name "%s" '
+                       u'already exists' % hostgroup1.cn)):
+            command()
+
+
+class TestAutomemberFindNegative(XMLRPC_test):
+
+    def test_find_with_nonexistent_group(self, automember_group, group1):
+        """ Try to find a rule with non-existent group """
+        group1.ensure_missing()
+        group1.find()
+        command = automember_group.make_find_command(type=u'group')
+        with raises_exact(errors.NotFound(
```

Comment:
automember-find does not raise exception when no result is found. The code should rather check that the result contains the following summary:
```
result = command()
assert_deepequal(dict(
            count=0,
            truncated=False,
            summary=u'0 rules matched',
            result=[],
        ), result)
```